src/builtins_types.rs

```diff
@@ -435,32 +436,33 @@ fn builtin_str_to_float(
     Err(LispError::new("str->float: requires a string"))
 }
 
-fn builtin_int_to_float(
-    environment: &mut Environment,
-    args: &mut dyn Iterator<Item = Expression>,
-) -> Result<Expression, LispError> {
-    if let Some(arg) = args.next() {
-        if args.next().is_none() {
-            if let ExpEnum::Int(i) = &eval(environment, arg)?.get().data {
-                return Ok(Expression::alloc_data(ExpEnum::Float(*i as f64)));
-            }
-        }
-    }
-    Err(LispError::new("int->float: requires an int"))
+/// Usage: (int->float int) -> float
+///     Cast an int as a float.
+/// Section: type
+/// Example:
+///     (test::assert-equal 0 (int->float 0))
+///     (test::assert-equal 10 (int->float 10))
+///     (test::assert-equal -101 (int->float -101))
+///     (test::assert-error (int->float "not int"))
+#[sl_sh_fn(fn_name = "int->float")]
+fn int_to_float(int: i64) -> f64 {
```

Comment:
I should look at the macro but being lazy.  If this could cause an error then can you return a Result here and have that become a Lisp error?  Out of scope for this PR but we should review these number routines for precision- it might make more sense to use verify the conversions are valid and error out of not (I think there is a TryFrom for these).  This one may always be safe but f64 to i64 might not be.

Reply:
addressed in the other PR